File: javascript/네트워크.js
function solution(n, computers) {//n 컴퓨터 개수 , computers relation matirx
    var answer = 0;
    for(let i=0;i<n;i++)computers[i][i]=0;
    let v = Array(n).fill(false);
    for(let i=0;i<n;i++){
        if(v[i]===true)continue;
        answer++;
        v[i]=true;
        go(i);
    }

    function go(idx){
        for(let i=0;i<n;i++){
            if(computers[idx][i]===1&&v[i]===false){
                v[i]=true;
                go(i);
            }
        }
    }
    return answer;
}

File: java/네트워크.java
import java.util.*;
class Solution {
    boolean visit[];
    int N;
    public int solution(int n, int[][] computers) {
        N=n;
        visit = new boolean[n];
        for(int i=0;i<n;i++){
            visit[i]=false;
        }
        int answer = 0;
        
        Queue<Integer> q = new LinkedList<Integer>();
        for(int i=0;i<n;i++){
            if(!visit[i]){
                answer++;
                q.add(i);
                while(!q.isEmpty()){
                    int cur = q.poll();
                    for(int j=0;j<n;j++){
                        if(computers[cur][j]==1 && !visit[j]){
                            visit[j]=true;
                            q.add(j);
                        }
                    }
                }
            }
        }
        
        return answer;
    }
}